src/accelerate/utils/operations.py

```diff
@@ -669,7 +669,7 @@ def _pad_across_processes(tensor, dim=0, pad_index=0, pad_first=False):
         old_size = tensor.shape
         new_size = list(old_size)
         new_size[dim] = max_size
-        new_tensor = tensor.new_zeros(tuple(new_size)) + pad_index
+        new_tensor = (tensor.new_zeros(tuple(new_size)) + pad_index).to(tensor.dtype)
```

Comment:
Hmm, I'm wondering how safe this is in case that the tensor dtype cannot represent the new data. E.g. when `pad_index` is not 0 or 1, casting this to `bool` will result in a loss of information.

Reply:
+1, `pad_index` in the context of LLMs is usually `-100`.

Reply:
I don't think there's any loss of useful information per se, given that the data is assigned directly downstream 
https://github.com/huggingface/accelerate/blob/c0552c9012a9bae7f125e1df89cf9ee0b0d250fd/src/accelerate/utils/operations.py#L679

What this does change though is the actual pad value. Anything non-zero (e.g., -100) will result in padding with `True` for `bool`.

Does it make sense to always pad with `False` for `bool`?